### PR TITLE
Optimize dandiset owner PUT endpoint

### PR DESCRIPTION
**Pull request by mvandenburgh**
Now that django-allauth uses a regular JSONField for `SocialAccount.extra_data`, we can optimize the dandisets owner PUT endpoint to be more efficient and do less SQL queries. See release notes for allauth here https://github.com/pennersr/django-allauth/blob/main/ChangeLog.rst#0580-2023-10-26

### CODE REVIEW

**Review by AlmightyYakob — APPROVED** [review]
LGTM

**Comment by mvandenburgh** [comment]
Looking at this again, the original ORM expression I wrote can be simplified - see https://github.com/dandi/dandi-archive/pull/1737/commits/5cb05e6a227c3305743e002ec3a2d39f4807149a. I tested it and it works just as well as the previous query.

**Comment by AlmightyYakob** [comment]
@mvandenburgh I noticed that we're still calling the `get_user_or_400` query `n` many times, which can be avoided. I pushed [`f3b362c`](https://github.com/dandi/dandi-archive/pull/1737/commits/f3b362c08f305c84701a4418a1c2da3acaea2028) which does it in one query (two if you include the `count` check). It also removes the check about the empty `owners` list, since I'm not even sure how that could be triggered (a 400 would be raised before that in any case).

Let me know what you think. Sorry for the trampling, it wasn't possible to achieve the whole diff with github suggestions.

File: dandiapi/api/views/dandiset.py [hunk 1]
```diff
@@ -372,25 +371,33 @@ def users(self, request, dandiset__pk):
...
+            # Ensure not all owners removed
+            if not serializer.validated_data:
                 raise ValidationError('Cannot remove all draft owners')
 
+            # Get all owners that have the provided username in one of the two possible locations
+            usernames = [owner['username'] for owner in serializer.validated_data]
+            user_owners = list(User.objects.filter(username__in=usernames))
+            socialaccount_owners = list(
+                SocialAccount.objects.select_related('user').filter(
+                    Q(extra_data__login__in=usernames)
```
Comment (mvandenburgh) [inline dandiapi/api/views/dandiset.py:383]: This Q object should no longer be needed now that there's only one condition.

**Review by AlmightyYakob — APPROVED** [review]

**Pull request merged**